code/game/machinery/computer/cloning.dm

```diff
@@ -462,6 +462,10 @@
 		scantemp = "<font class='average'>Subject already in database.</font>"
 		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
 		return
+	if(mob_occupant.has_trait(TRAIT_NOSCAN))
+		scantemp = "<font class='bad'>Subject has no DNA, or has DNA that cannot be scanned.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+		return

```

Comment:
this trait seems to cover the same cases as TRAIT_NOCLONE (aka NODNA). Since NOCLONE is barely used you could conflate them into a proper TRAIT_NODNA.

Reply:
agree in principle with XDTM here

Reply:
actually impossible, NOCLONE organics can be scanned if scan_level >= 2, probably something stockpart related without probing more code. Thus NOCLONE and NOSCAN must be seperate.

Reply:
That's stock part related, yes. I can only think of ling absorbs in which NOCLONE has any relevance though.

You could just suck it up and make them clonable post-upgrades. Lore wise an upgraded cloner doubles as highly-specialized robotics fabricator or something.

Not entirely a fan of having so many flavors of "PLEASE DONT COME BACK INTO THE ROUND" with very small differences.

Reply:
Actually the NOCLONE check for cloning could easily be replaced by a HUSK check. Lings still husk corpses, space does as well, NOCLONE (NODNA) can be kept completely unscannable.

Reply:
>add noclone for balance
>science adds a way to bypass it

amazing

Reply:
>noclone as balance
>codebase has veered away from permanent death for the past 3 years
>sheeple blame science

amazing

Reply:
how the hell would you know what the codebase has and hasn't veered away from